scripts/sipify.py

```diff
@@ -2218,11 +2225,20 @@ def detect_non_method_member(line):
     if match:
         CONTEXT.current_line = f"{match.group(1)}{match.group(3)};"
 
-    pattern = r'^\s*(?:const |virtual |static |inline )*(?!explicit)([(?:long )\w:]+(?:<.*?>)?)\s+(?:\*|&)?(\w+|operator.{1,2})\(.*$'
+    pattern = r'^\s*((?:const |virtual |static |inline ))*(?!explicit)([(?:long )\w:]+(?:<.*?>)?)\s+(?:\*|&)?(\w+|operator.{1,2})\(.*$'
     match = re.match(pattern, CONTEXT.current_line)
     if match:
-        CONTEXT.current_method_name = match.group(2)
-        return_type_candidate = match.group(1)
+        CONTEXT.current_method_name = match.group(3)
+        return_type_candidate = match.group(2)
+        is_static = bool(match.group(1) and 'static' in match.group(1))
+        class_name = CONTEXT.current_fully_qualified_class_name()
+        if CONTEXT.current_method_name in CONTEXT.static_methods[class_name]:
```

Comment:
Why not just store only the static methods in a set instead of a dict which contains actually all methods although we need only the static ones?

Reply:
@troopa81 so we can detect if the same method name is used for both static and non static C++ methods. If so, then we reset this flag to false and can't use the wrapper.